### PR TITLE
Rewrite savewrong.deno test to Rust.

### PR DESCRIPTION
**Pull request by BearLemma**

### CODE REVIEW

File: cli/tests/integration_tests/framework.rs [hunk 1]
```diff
@@ -550,6 +550,20 @@ impl Response {
...
+            "Unexpected text response for HTTP {} {}\nResponse status {}, body {:?}, expected {:?}",
+            self.method,
+            self.url,
+            self.status,
+            actual,
+            expected,
+        );
+        self
+    }
+
```
Comment (MarinPostma) [inline cli/tests/integration_tests/framework.rs:566]: uhm you probably want to make that a macro, so the error is shown a the actual call points, or return an error from the function, and panic in the context of the test.

Comment (MarinPostma) [reply cli/tests/integration_tests/framework.rs:566]: then I think that returning an error and letting the caller unwrap is a good option?

Comment (MarinPostma) [reply cli/tests/integration_tests/framework.rs:566]: just replace the assert with `anyhow::ensure!`

Comment (MarinPostma) [reply cli/tests/integration_tests/framework.rs:566]: ok ok, your realm your rules 😄

**Review by MarinPostma — APPROVED** [review]

File: cli/tests/integration_tests/framework.rs (hunk 1 above)
Comment (honzasp) [reply cli/tests/integration_tests/framework.rs:566]: The point here is that you get all the information about the response (status, body, etc.) in the error message, so this is vastly superior to just `assert!(response.text().contains(expected))`. Also, this being a method, these calls can "fluently" chain.

Comment (honzasp) [reply cli/tests/integration_tests/framework.rs:566]: (We could perhaps to the same magic that `unwrap()` uses to print the location of the caller?)

File: cli/tests/integration_tests/rust_tests/auth.rs [hunk 2]
```diff
@@ -75,3 +75,85 @@ pub async fn next_auth_crud(mut c: TestContext) {
...
+
+    c.chisel
+        .post("/dev/auth_users")
+        .json(json!({"name":"Foo", "email":"foo@t.co"}))
+        .send()
+        .await
+        .assert_status(500)
+        .assert_text_contains("Error: Cannot save into type AuthUser.");
+    c.chisel
+        .post("/dev/auth_users/abcd-1234-eeee-5678")
```
Comment (dekimir) [inline cli/tests/integration_tests/rust_tests/auth.rs:100]: This was a PUT originally.

**Pull request merged**